source/MaterialXRuntime/Private/PvtPrim.cpp

```diff
@@ -117,6 +117,21 @@ void PvtPrim::removeRelationship(const RtToken& name)
     }
 }
 
+void PvtPrim::renameRelationship(const RtToken& name, const RtToken& newName)
+{
+    if (getRelationship(newName))
+    {
+        throw ExceptionRuntimeError("A relationship named '" + newName.str() + "' already exists in prim '" + getName().str() + "'");
```

Comment:
Is it the callers responsibility to make sure the name is unique? Or should this API find the next available name?

Reply:
Good point. I could change the behavior to allow name collisions and resolve them internally, as we do when renaming prims in a stage.